packages/react/src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -649,6 +673,9 @@ const MultiSelect = React.forwardRef(
       selectedItems.length > 0 &&
       selectedItems.map((item) => (item as selectedItemType)?.text);
 
+    const selectedItemsWithoutSelectAll = selectedItems.filter(
+      (item: any) => item.id !== 'select-all-option'
+    );
```

Comment:
no we are using a new property in the `selectAll` object, called as `selectAllFlag` which just a boolean so that we don't mess with id being same as user's data.